### PR TITLE
use cocina-models factories for specs

### LINKED ISSUE #887: Use cocina factory for spec
To reduce boilerplate.

See https://github.com/sul-dlss/dor-services-app/pull/3948

### PR DESCRIPTION
**Pull request by ndushay**
You might want to review the commits separately

## Why was this change made? 🤔

to reduce boilerplate in our specs.

Fixes #887 

## How was this change tested? 🤨

specs

⚡ ⚠ If this change has cross service impact, including writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: spec/lib/dor/structural_metadata_spec.rb [hunk 1]
```diff
@@ -51,21 +51,7 @@
 
     let(:apo_druid) { 'druid:pp000pp0000' }
```
Comment (justinlittman) [inline spec/lib/dor/structural_metadata_spec.rb:52]: `apo_druid` no longer necessary.

File: spec/robots/assembly/checksum_compute_spec.rb [hunk 2]
```diff
@@ -4,7 +4,8 @@
 
 RSpec.describe Robots::DorRepo::Assembly::ChecksumCompute do
   let(:robot) { described_class.new }
-  let(:druid) { 'aa222cc3333' }
+  let(:druid) { 'bb222cc3333' }
```
Comment (justinlittman) [inline spec/robots/assembly/checksum_compute_spec.rb:7]: For what it is worth, in other places we use `druid` and `bare druid`.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

Doesn't it feel good to throw code away?

File: spec/robots/assembly/checksum_compute_spec.rb (hunk 2 above)
Comment (ndushay) [reply spec/robots/assembly/checksum_compute_spec.rb:7]: true.  fixed.

**Pull request merged**